client/packrat-ui/form/index.js

```diff
@@ -62,46 +90,76 @@ const ReusableForm = forwardRef((props, ref) => {
     inputRefs.current[name]?.focus();
   };
 
-  useImperativeHandle(ref, () => ({
-    focus: focusInput,
-  }));
+  // useImperativeHandle(ref, () => ({
+  //   focus: focusInput,
+  // }));
+
+  useEffect(() => {
+    const subscription = watch((data) => {
+      try {
+        schema.parse(data);
+        clearErrors();
+        setIsValidated(false);
+      } catch (error) {
+        trigger();
+      }
+    });
+    return () => subscription.unsubscribe();
+  }, [watch]);
 
   return (
-    <Form onSubmit={handleSubmit(onSubmit)}>
+    <Form onSubmit={() => handleSubmit(onSubmit)}>
       {fields.map((field) => (
-        <XStack
-          overflow="hidden"
-          space="$2"
-          margin="$3"
-          padding="$2"
-          key={field.name}
-        >
-          {field.label && <Label htmlFor={field.name}>{field.label}</Label>}
+        <XStack overflow="hidden" key={field.name} flexDirection="column">
+          {field.label && (
+            <Label
+              htmlFor={field.name}
+              color="gray"
+              fontSize={15}
+              _dark={{
+                color: 'warmGray.200',
+              }}
+            >
+              {field.label}
+            </Label>
+          )}
           <YStack space="$1">
             <Controller
               name={field.name}
               control={control}
-              render={({ field: fieldProps }) => (
+              render={({ field: fieldProps, fieldState: { error } }) => (
                 <RenderInput
                   field={field}
+                  control={control}
+                  error={error}
                   fieldProps={{
                     ...fieldProps,
                     ref: (el) => (inputRefs.current[field.name] = el),
                   }}
                 />
               )}
             />
-            <RenderError
-              error={field.errorMessage}
-              fieldError={errors[field.name]?.message}
-            />
             <RenderHelperText text={field.helperText} />
           </YStack>
         </XStack>
       ))}
-      <Form.Trigger asChild>
-        <Button>Submit</Button>
-      </Form.Trigger>
+      {children}
+      {onSubmit && (
+        <Form.Trigger asChild>
+          <VStack alignItems={'center'}>
+            <Button
+              mt="2"
+              colorScheme={'indigo'}
+              alignContent={'center'}
+              type="submit"
+              onPress={handleSubmit(onSubmit)}
+              // disabled={isValidated}
```

Comment:
we should disable if not validated 

- [ ]

Reply:
i will but the button styling isnt really working